src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,188 @@
+            } catch (DateTimeParseException e) {
+                throw new InvalidTaskException("Invalid date format! [yyyy-mm-dd] Eg. [2019-12-01]");
+            }
+
+            break;
+
+        case EVENT:
+            if (taskArr.length <= 1) {
+                throw new InvalidTaskException("☹ OOPS!!! The description of a event cannot be empty.");
+            }
+            params = taskArr[1].split(" /at ");
+            if (params.length <= 1) {
+                throw new InvalidTaskException("☹ OOPS!!! The time of an event cannot be empty.");
+            }
+            try {
+                LocalDate date = LocalDate.parse(params[1]);
+                task = new duke.tasks.Event(params[0], date);
+            } catch (DateTimeParseException e) {
+                throw new InvalidTaskException("Invalid date format! [yyyy-mm-dd] Eg. [2019-12-01]");
+            }
+            break;
+
+        default:
+            throw new InvalidTaskException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+        return task;
+    }
+
+    /**
+     * Convert a string from csv format to task.
+     *
+     * @param taskString csv string of task
+     * @param delimiter  separator
+     * @return task object
+     * @throws InvalidTaskException invalid task
+     */
+    public static Task fromCsv(String taskString, String delimiter) throws InvalidTaskException {
+        Task task;
+        String[] taskArr = taskString.split(delimiter);
+
+        if (taskArr.length < 3) {
+            throw new InvalidTaskException("Insufficient values");
+        }
+
+        String taskCode = taskArr[0];
+        boolean taskCompleted = Objects.equals(taskArr[1], "x");
+        String taskDescription = taskArr[2];
+
+        switch (taskCode) {
+        case duke.tasks.Todo.TASK_CODE:
+            task = new duke.tasks.Todo(taskDescription);
+            break;
+        case Deadline.TASK_CODE:
+            if (taskArr.length < 4) {
+                throw new InvalidTaskException("Insufficient values");
+            }
+            LocalDate by = LocalDate.parse(taskArr[3]);
+            task = new Deadline(taskDescription, by);
+            break;
+        case duke.tasks.Event.TASK_CODE:
+            if (taskArr.length < 4) {
+                throw new InvalidTaskException("Insufficient values");
+            }
+            LocalDate at = LocalDate.parse(taskArr[3]);
+            task = new duke.tasks.Event(taskDescription, at);
+            break;
+        default:
+            throw new InvalidTaskException("Task format is invalid");
+        }
+
+        if (taskCompleted) {
+            task.markAsCompleted();
+        }
+        return task;
+    }
+
+    /**
+     * Checks if the task description contains the keyword.
+     *
+     * @param keyword String to search for
+     * @return true if the description contains the keyword
+     */
+    public boolean matches(String keyword) {
```

Comment:
Perhaps you can change the method name to
`has_contains_keyword(String keyword)`
so that it sounds more like boolean